### PR TITLE
Throw MisconfigurationException when accelerator is available but dev…

### PR DESCRIPTION
**Pull request by kaushikb11**
…ices value is not supported

## What does this PR do?

Fixes #\<issue_number>

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/trainer/connectors/accelerator_connector.py [hunk 1]
```diff
@@ -435,18 +435,20 @@ def _map_deprecated_devices_specfic_info_to_accelerator_and_device_flag(
...
                 rank_zero_warn(
                     f"The flag `devices={devices}` will be ignored, "
                     f"instead the device specific number {deprecated_devices_specific_flag} will be used"
                 )
 
-            if [(num_processes is not None), (gpus is not None), (ipus is not None), (tpu_cores is not None)].count(
-                True
-            ) > 1:
+            if len(deprecated_devices_specific_flags) > 1:
                 # TODO: @awaelchli improve error message
```
Comment (carmocca) [inline pytorch_lightning/trainer/connectors/accelerator_connector.py:452]: if you have `deprecated_device_specific_flags` be a dictionary, you could resolve this TODO because we can still check the length of it, and extract the keys from the dictionary and show them in the error message

**Comment by carmocca** [comment]
Nvm, not a breaking change because the user gets an error anyways :+1:

File: pytorch_lightning/trainer/connectors/accelerator_connector.py [hunk 2]
```diff
@@ -494,6 +496,11 @@ def _set_parallel_devices_and_init_accelerator(self) -> None:
                 )
             self.accelerator = AcceleratorRegistry.get(self._accelerator_flag)
 
+        if self._devices_flag in ([], 0, "0"):
+            raise MisconfigurationException(
+                f"`{self._devices_flag}` value for devices is not valid for {self.accelerator.__class__.__qualname__}"
```
[the remark this answers is not in the document]
Comment (kaushikb11) [reply pytorch_lightning/trainer/connectors/accelerator_connector.py:501]: Umm, I agree with the thought. But it could be `Trainer(gpus=x)` as well.

[the remark this answers is not in the document]
Comment (carmocca) [reply pytorch_lightning/trainer/connectors/accelerator_connector.py:501]: Considering that gpus is going to be deprecated hard-coding `devices` in the message should be nice.

Or you can add logic to set the correct argument if you prefer

**Comment by DuYicong515** [comment]
> Nvm, not a breaking change because the user gets an error anyways

This is a breaking change, since before it falls back to CPU and now it throws errors. 

I think it might be more common to use `Trainer(gpus=0)` compared to using `Trainer(accelerator="gpu", devices=0)`.  Here are some [examples](https://sourcegraph.com/search?q=context:global+gpus%3D.*+if+torch.cuda.is_available%28%29+else+0&patternType=regexp) I see. It's also in [lightning-tutorials]( https://github.com/PyTorchLightning/lightning-tutorials/blob/1143a73d47fc89acf53f404a25844a6f08927534/course_UvA-DL/09-normalizing-flows/NF_image_modeling.py#L920). Since `gpus` are going to be deprecated in Trainer args, I agree with the change. But let's still highlight it's a breaking change.

**Comment by carmocca** [comment]
@DuYicong515 If I'm not mistaken, #12410 (included in 1.6) was the breaking change (and tagged as such) but missed raising a proper exception for `devices=0`.

However it does still raise an exception (`TypeError`), so in that sense, this is not a breaking change but a UX improvement.

**Comment by DuYicong515** [comment]
@carmocca This PR has 2 changes
1/ `Trainer(accelerator=xpu, devices=0/"0"/[])` The PR improves the exceptions being raised, which is an UX improvements
2/ `Trainer(gpus/ipus/tpu_cores=0/"0"/[])` It used to fall back to CPU. After this PR, it starts to throw exceptions. And that's a breaking change.

**Comment by carmocca** [comment]
Closing as it's stale, keeping the linked issue open
